test_runner/src/main/kotlin/ftl/ios/xctest/XcTestData.kt

```diff
@@ -54,15 +54,23 @@ private fun emptyXcTestRunData() = XcTestRunData(
     nsDict = NSDictionary(),
     version = V1
 )
+private fun IosArgs.filterTestConfigurationsIfNeeded(
+    configurations: Map<String, Map<String, List<String>>>
+): Map<String, Map<String, List<String>>> {
+    val only = onlyTestConfiguration ?: return configurations.filterKeys { it != skipTestConfiguration }
+    return configurations.filterKeys { it == only }
+}
```

Comment:
```suggestion
private fun IosArgs.filterTestConfigurationsIfNeeded(
    configurations: Map<String, Map<String, List<String>>>
): Map<String, Map<String, List<String>>> = when {
    onlyTestConfiguration != null -> configurations.filterKeys { it == onlyTestConfiguration }
    skipTestConfiguration != null -> configurations.filterKeys { it != skipTestConfiguration }
    else -> configurations
}
```

In most cases, even one return is too much :).